AgContext.yaml

```diff
@@ -0,0 +1,321 @@
+type: object
+required:
+  - id
+  - crop_type
+  - crop_stage
+  - soil_colour
+  - surface_cover
+  - surface_coverage
+  - weather_description
+  - location_lat
+  - location_long
+  - location_datum
+  - camera_model
+  - lens_model
+  - camera_height
+  - camera_angle
+  - camera_fov
+  - photography_description
+  - cropped_to_plant
+
+properties:
+
+  id:
+    type:
+      number
+    description: |-
+        A number identifying the AgContext.
+
+  crop_type:
+    type:
+      string
+    description: |-
+          General crop type.
+          A string describing the general cropping scenario (e.g. "root vegetable").
+
+  grains_descriptive_text:
+    type:
+      string
+    enum:
+      - emergence
+      - seedling
+      - tillering
+      - stem_elongation
+      - booting
+      - ear_emergence
+      - flowering
+      - milky_dough
+      - dough
+      - ripening
+      - na
+    description: |-
+        Grains descriptive text.
+        One of ten possible strings describing the crop developmental stage.
+        If this AgContext is not in a cropping environment, use value "na".
+
+  bbch_descriptive_text:
```

Comment:
I think we will have this available in a data entry form, but shouldn't need it as a separate property in the JSON schema. I don't see anything wrong with having a `value_descriptions` or `bbch_descriptions` in the definition of `bbch_code` that could be used to construct data entry forms or search interfaces... But maybe we're getting too generic then and extending JSON schema beyond its purpose.